adobe/react-spectrum
# fix(#2675): TableView: when renderEmptyTableState select all throws if selected from kb

Closes #2675 

As @snowystinger suggest, issue is cause from undefined `firstItem` which is cause from undefined value from `getFirstKey()`.

So, I check nullish `getFirstKey()` on `getKeyBelow()` and also check nullish `key` on `getKeyForSearch()` too.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/2675).

## 📝 Test Instructions:

In storybook path `/story/tableview--crud `

- Remove all items 
- Click the select al checkbox 
- Hit spacebar 
  expected no error on console

## 🧢 Your Project:

RSP
